### PR TITLE
update electron to 1.7.5

### LINKED ISSUE #1003: Update electron type declarations to add popup menu method with new async arg
See https://github.com/desktop/desktop/pull/998 for discussion.

Once 1.6.2 docs are published, submit a PR for [this signature](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/electron/index.d.ts#L2626-L2633) to be updated to reflect whatever's currently required.

Looking at the source for `popup`, I see two different methods:

 - `menu.popup(x, y, positioningItem)` - the current one (doesn't support `positioningItem`)
 - `menu.popup(window, { })` - a new one, which supports optional `x`, `y`, `positioningItem` and `async` arguments

### LINKED ISSUE #1502: Toggling Full screen off causes the app to crash and start in full screen afterwards
### Description

Hitting F11 will resize GitHub to use the full screen, however, hitting it again will cause the the app to resize but only show a black screen. Clicking will cause the app to crash but it will continue to run in the Task Manager, resulting in GitHub not opening.

Once the process is killed in the Task Manager, GitHub can be started again but will open in full screen. Repeat.

### Version

**GitHub Desktop version:** 0.5.3

**OS version:** Windows 10 version 1703

### Steps to Reproduce

1. Open GitHub
2. Toggle Full screen mode on
3. Toggle Full screen mode off

**Expected behaviour:** GitHub shows a normal window again

**Actual behaviour:** The app becomes fully black, any further interaction with any window will cause the black window to disappear.

**Reproduces how often:** 100%

### Logs

````
2017-05-16T17:52:20.724Z - error: Uncaught exception on main process
````

### LINKED ISSUE #2607: Opens a new chrome tab on ]
### Description

With hungarian layout pressing the `]` character is done with altgr+G which the Github application catches as ctrl+alt+g which opens a chrome tab.

### Version

**GitHub Desktop version:** 0.8.0

**OS version:** Microsoft Windows [Version 10.0.15063]

### Steps to Reproduce

1. Press alt gr+G

**Expected behavior:** It does not open a chrome tab.

**Actual behavior:** Opens a chrome tab.

**Reproduces how often:** Every time :)

### PR DESCRIPTION
**Pull request by shiftkey**
Fixes #1003
Fixes #2607
Fixes #1502

Opening this up to start testing the latest stable version of Electron:

 - [x] evaluate transition from 1.6.x  for any potential breakage

### CODE REVIEW

**Comment by shiftkey** [comment]
Pro Tip: if you run `npm run build:prod && npm run start:prod` it will affect your existing installation, and subsequently running the installed `1.6.x` app won't work. Not sure the best way to manage this for situations like CI builds, let's see what breaks.

**Comment by joshaber** [comment]
>confirm #2323 is part of this update

I _think_ that'll only be fixed in the 1.8.x series, but I'm not totally sure.

https://github.com/desktop/desktop/issues/1502 should hopefully be fixed 🙏

**Comment by shiftkey** [comment]
> I think that'll only be fixed in the 1.8.x series, but I'm not totally sure.

Looking at the packaged bits between `1.6.11` and `1.7.5`, it looks like the same version of Squirrel.Mac is bundled.

**Comment by shiftkey** [comment]
Opening this up for review and investigating whether this should go in before the 1.0 update.

**Comment by joshaber** [comment]
I'm ok with landing this before 1.0, unless anyone else has objections /cc @desktop/core

File: app/src/main-process/main.ts [hunk 1]
```diff
@@ -238,10 +238,7 @@ app.on('ready', () => {
       }
 
       const window = BrowserWindow.fromWebContents(event.sender)
-      // TODO: read https://github.com/desktop/desktop/issues/1003
-      // to clean up this sin against T Y P E S
-      const anyMenu: any = menu
```
Comment (joshaber) [inline app/src/main-process/main.ts]: 🤘

**Pull request merged**